### PR TITLE
[test/plugin] to check agent register status

### PR DESCRIPTION
**Pull request by dmsolr**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3698 

1. curl time-out, default 2s
2. visit scenario's entry after agent register success.

### CODE REVIEW

File: test/plugin/mock-collector/src/main/java/org/apache/skywalking/plugin/test/mockcollector/Main.java [hunk 1]
```diff
@@ -47,6 +55,18 @@ public static void main(String[] args) throws Exception {
         String contextPath = "/";
         ServletContextHandler servletContextHandler = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
         servletContextHandler.setContextPath(contextPath);
+        servletContextHandler.addServlet(new ServletHolder(new HttpServlet() {
+            @Override
+            protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+                if (ValidateData.INSTANCE.getRegistryItem().getApplications().isEmpty()) {
+                    resp.setStatus(500);
```
Comment (arugal) [inline test/plugin/mock-collector/src/main/java/org/apache/skywalking/plugin/test/mockcollector/Main.java]: Is there a case where there are multiple ```agent``` in one test case.

File: test/plugin/scenarios/ehcache-2.x-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -23,26 +23,8 @@ registryItems:
   heartbeat: []
 segmentItems:
   - applicationCode: ehcache-2.x-scenario
-    segmentSize: 2
+    segmentSize: ge 2
```
Comment (arugal) [inline test/plugin/scenarios/ehcache-2.x-scenario/config/expectedData.yaml]: It is recommended to use ```gt 1```

**Comment by dmsolr** [comment]
> Can you explain why change `<n>` to `ge <n>` in `expectedData.yml`, does it have side effects?

This is why we modify the `health-check` mechanism. The `healthcheck` API will be called many times when health-check failed.

And then I set `curl` timeout which intends to present `the failure` more easy.

**Comment by dmsolr** [comment]
In fact, I think the validate-tool skips the segment of `health-check`.  HDYT?

**Comment by kezhenxu94** [comment]
> In fact, I think the validate-tool skips the segment of `health-check`. HDYT?

Ignoring the `healthCheck` span sounds good to me, health check is not related to the plugin itself, in fact, E2E ignore the health check too IIRC

File: test/plugin/mock-collector/src/main/java/org/apache/skywalking/plugin/test/mockcollector/Main.java (hunk 1 above)
Comment (wu-sheng) [reply test/plugin/mock-collector/src/main/java/org/apache/skywalking/plugin/test/mockcollector/Main.java]: Mutiple agent is possible, but let's avoid it if possible.

**Comment by wu-sheng** [comment]
Please remove the socket.io test for now.

**Pull request merged**